code/modules/paperwork/pen.dm

```diff
@@ -339,3 +339,52 @@
 	force = 5
 	wound_bonus = 100
 	demolition_mod = 9000
+
+// screwdriver pen!
+
+/obj/item/pen/screwdriver
+	desc = "A pen with an extendable screwdriver tip. This one has a yellow cap."
+	icon_state = "pendriver"
+	toolspeed = 1.2  // gotta have some downside
+	/// whether the pen is extended
+	var/extended = FALSE
+
+/obj/item/pen/screwdriver/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/transforming, \
+		throwforce_on = 5, \
+		w_class_on = WEIGHT_CLASS_SMALL, \
+		sharpness_on = TRUE)
+
+	RegisterSignal(src, COMSIG_TRANSFORMING_ON_TRANSFORM, .proc/toggle_screwdriver)
+	AddElement(/datum/element/update_icon_updates_onmob)
+
+
+/obj/item/pen/screwdriver/vv_edit_var(var_name, var_value)
+	if(var_name == NAMEOF(src, extended))
+		if(var_value != extended)
+			attack_self()
```

Comment:
toggle_screwdriver instead?

Reply:
then the transform component is completely skipped, and toggle_screwdriver does nothing besides toggle a few vars that arent managed by transform

Reply:
sec